tests/qos/test_pfc_pause.py

```diff
@@ -0,0 +1,348 @@
+                           send_pause=send_pause,
+                           pfc_pause=pfc_pause,
+                           pause_prio=pause_prio,
+                           pause_time=pause_time, 
+                           max_test_intfs_count=max_test_intfs_count)
+            
+    else:
+        return None 
+    
+def test_pfc_pause_lossless(ansible_adhoc,
+                            testbed, 
+                            conn_graph_facts, 
+                            leaf_fanouts, lossless_prio_dscp_map):
+    
+    """ @Summary: Test if PFC pause frames can pause a lossless priority without affecting the other priorities """    
+    setup_testbed(ansible_adhoc, testbed, leaf_fanouts)
+
+    errors = []
+    
+    """ DSCP vlaues for lossless priorities """
+    lossless_dscps = [int(dscp) for prio in lossless_prio_dscp_map for dscp in lossless_prio_dscp_map[prio]]
+    """ DSCP values for lossy priorities """
+    lossy_dscps = list(set(range(64)) - set(lossless_dscps))
+    
+    for prio in lossless_prio_dscp_map:
+        """ DSCP values of the other lossless priorities """
+        other_lossless_dscps = list(set(lossless_dscps) - set(lossless_prio_dscp_map[prio]))
+        """ We also need to test some DSCP values for lossy priorities """
+        other_dscps = other_lossless_dscps + random.sample(lossy_dscps, k=2)
+        
+        for dscp in lossless_prio_dscp_map[prio]:
+            for dscp_bg in other_dscps:
+                results = run_test(ansible_adhoc=ansible_adhoc, 
+                                   testbed=testbed,
+                                   conn_graph_facts=conn_graph_facts,
+                                   leaf_fanouts=leaf_fanouts, 
+                                   dscp=dscp,
+                                   dscp_bg=dscp_bg,
+                                   queue_paused=True,
+                                   send_pause=True,
+                                   pfc_pause=True,
+                                   pause_prio=prio,
+                                   pause_time=65535,
+                                   max_test_intfs_count=MAX_TEST_INTFS_COUNT)
+
+                """ results should not be none """
+                if results is None:
+                    assert 0 
+            
+                errors = dict()
+                for intf in results:
+                    if len(results[intf]) != 2:
+                        continue
+                
+                    pass_count = results[intf][0]
+                    total_count = results[intf][1]
+
+                    if total_count == 0:
+                        continue
+            
+                    if pass_count < total_count * PTF_PASS_RATIO_THRESH:
+                        errors[intf] = results[intf]
+
+                if len(errors) > 0:
+                    print "errors occured:\n{}".format("\n".join(errors))
+                    assert 0 
+
+def test_no_pfc(ansible_adhoc,
```

Comment:
Please update the PR description to include this test as well

Reply:
Fixed